src/controllers/auth.controller.js

```diff
@@ -0,0 +1,131 @@
+import { User } from '../models/user.js';
+import { userServices } from '../services/user.service.js';
+import { jwtService } from '../services/jwt.service.js';
+import { ApiError } from '../exceptions/api.error.js';
+import bcrypt from 'bcrypt';
+import { tokenService } from '../services/token.service.js';
+
+function validateEmail(value) {
+  if (!value) {
+    return 'Email is required';
+  }
+
+  const emailPatterns = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+
+  if (!emailPatterns.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+function validatePassword(value) {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+}
+
+const register = async (req, res) => {
+  const { email, password } = req.body;
+
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+  if (errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
+  }
+
+  const hashedPassword = await bcrypt.hash(password, 10);
+  await userServices.register(email, hashedPassword);
+
+  res.send({ message: 'OK' });
+};
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+  const user = await User.findOne({
+    where: { activationToken: activationToken },
+  });
+
+  if (!user) {
+    res.sendStatus(404);
+    return;
+  }
+
+  user.activationToken = null;
+  user.save();
+
+  res.send(user);
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  const user = await userServices.findByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
```

Comment:
For authentication errors, it's better to use `ApiError.unauthorized` with a 401 status code instead of `badRequest`. This accurately reflects the nature of the error.